drgn_tools/rds.py

```diff
@@ -1020,6 +1117,421 @@ def rds_sock_info(
+                        conn_raddr,
+                        rport_rm,
+                        conn_tos,
+                        seq,
+                        num_bytes,
+                    ]
+                )
+
+    if queue.lower() == "send":
+        print("Send Message Queue:")
+    elif queue.lower() == "retrans":
+        print("Retransmit Message Queue:")
+
+    print_table(msg_queue, outfile, report)
+    print("\n")
+    return None
+
+
+def rds_print_msg_queue(
+    prog: drgn.Program,
+    queue: str = "All",
+    laddr: Optional[str] = None,
+    raddr: Optional[str] = None,
+    tos: Optional[str] = None,
+    lport: Optional[str] = None,
+    rport: Optional[str] = None,
+    ret: Optional[bool] = False,
+    outfile: Optional[str] = None,
+    report: bool = False,
+) -> None:
+    """
+    Print the rds msg queue similar rds-info -srt
+
+    :param prog: drgn program
+    :prarm queue: The msg queue to be displaed. Ex: 'send', 'recv' , 'retrans', 'All'. Default: 'All'
+    :param laddr: comma seperated string list of LOCAL-IP.  Ex: '192.168.X.X, 10.211.X.X, ...'
+    :param raddr: comma seperated string list of REMOTE-IP.  Ex: '192.168.X.X, 10.211.X.X, ...'
+    :param tos: comma seperated string list of TOS.  Ex: '0, 3, ...'
+    :param lport: comma seperated string list of lport.  Ex: 2259, 36554, ...'
+    :param rport: comma seperated string list of rport.  Ex: 2259, 36554, ...'
+    :param outfile: A file to write the output to.
+    :param report: Open the file in append mode. Used to generate a report of all the functions in the rds module.
+    :returns: None
+
+    """
+    msg = ensure_debuginfo(prog, ["rds"])
+    if msg:
+        print(msg)
+
+    if queue.lower() == "all":
+        rds_print_send_retrans_msg_queue(
+            prog, "send", laddr, raddr, tos, lport, rport, ret, outfile, report
+        )
+        rds_print_send_retrans_msg_queue(
+            prog,
+            "retrans",
+            laddr,
+            raddr,
+            tos,
+            lport,
+            rport,
+            ret,
+            outfile,
+            report,
+        )
+        rds_print_recv_msg_queue(
+            prog, laddr, raddr, tos, lport, rport, ret, outfile, report
+        )
+    elif queue.lower() == "send" or queue.lower() == "snd":
+        rds_print_send_retrans_msg_queue(
+            prog, "send", laddr, raddr, tos, lport, rport, ret, outfile, report
+        )
+    elif queue.lower() == "retrans" or queue.lower() == "re":
+        rds_print_send_retrans_msg_queue(
+            prog,
+            "retrans",
+            laddr,
+            raddr,
+            tos,
+            lport,
+            rport,
+            ret,
+            outfile,
+            report,
+        )
+    elif queue.lower() == "recv" or queue.lower() == "rcv":
+        rds_print_recv_msg_queue(
+            prog, laddr, raddr, tos, lport, rport, ret, outfile, report
+        )
+    else:
+        print("Incorrect queue arg provided !")
```

Comment:
Let's reduce this a bit:
```suggestion
    queue = queue.lower()
    if queue not in ("all", "send", "snd", "retrans", "re", "recv", "rcv"):
        print(f"Unknown queue type '{queue}'. Expected: all, send, retrans, or recv")
        return

    if queue in ("send", "snd", "all"):
        rds_print_send_retrans_msg_queue(
            prog, "send", laddr, raddr, tos, lport, rport, ret, outfile, report
        )
    if queue in ("retrans", "re", "all"):
        rds_print_send_retrans_msg_queue(
            prog,
            "retrans",
            laddr,
            raddr,
            tos,
            lport,
            rport,
            ret,
            outfile,
            report,
        )
    if queue in ("recv", "rcv", "all"):
        rds_print_recv_msg_queue(
            prog, laddr, raddr, tos, lport, rport, ret, outfile, report
        )
```